_observability-plugin/ppl/commands.md

```diff
@@ -831,3 +831,84 @@ PPL query:
 ```ppl
 search source=my_index | where match(message, "this is a test", operator=and, zero_terms_query=all)
 ```
+
+## ad
+
+The `ad` command applies Random Cut Forest (RCF) algorithm in ml-commons plugin on the search result returned by a PPL command.Based on the input, two types of RCF algorithms will be utilized: fixed in time RCF for processing time-series data, batch RCF for processing non-time-series data.
+
+### Fixed In Time RCF For Time-series Data Command Syntax
+
+```sql
+ad \<shingle\_size\> \<time\_decay\> \<time\_field\>
+```
+
+Field | Description | Required
+:--- | :--- |:---
+`shingle\_size` | A consecutive sequence of the most recent records. The default value is 8. | No
+`time\_decay` | Specifies how much of the recent past to consider when computing an anomaly score. The default value is 0.001. | No
+`time\_field` | Specifies the time filed for RCF to use as time-series data. | Yes
```

Comment:
Current limitation: the value of `time_field` should be long value (Timestamp in milliseconds), or String value following this date format "yyyy-MM-dd HH:mm:ss"